diff --git a/vendor/magento/module-cms/Model/PageRepository.php b/vendor/magento/module-cms/Model/PageRepository.php
index 7e84b93f..4bbcefe3 100644
--- a/vendor/magento/module-cms/Model/PageRepository.php
+++ b/vendor/magento/module-cms/Model/PageRepository.php
@@ -190,7 +190,11 @@ class PageRepository implements PageRepositoryInterface
                 $page->setStoreId($storeId);
             }
             $this->validateLayoutUpdate($page);
-            $this->validateRoutesDuplication($page);
+
+            if ($page->getIdentifier() != 'affirm') {
+                $this->validateRoutesDuplication($page);
+            }
+
             $this->resource->save($page);
             $this->identityMap->add($page);
         } catch (LocalizedException $exception) {
