diff --git a/vendor/magento/module-quote/Model/ResourceModel/Quote.php b/vendor/magento/module-quote/Model/ResourceModel/Quote.php
index 11d4be20..a912d70c 100644
--- a/vendor/magento/module-quote/Model/ResourceModel/Quote.php
+++ b/vendor/magento/module-quote/Model/ResourceModel/Quote.php
@@ -62,17 +62,24 @@ class Quote extends AbstractDb
         $select = parent::_getLoadSelect($field, $value, $object);
         $storeIds = $object->getSharedStoreIds();
 
+        $mainTable = $this->getMainTable();
+        $whereCondition = "store_id";
+
+        if ($mainTable) {
+            $whereCondition = $mainTable.".store_id";
+        }
+
         if ($storeIds) {
             // The comparison the arrays [0] != ['*'] returns `true` in PHP >= 8 and `false` otherwise
             // This check emulates an old behavior (as it was in PHP 7)
             if ($storeIds !== ['*'] && $storeIds !== [0]) {
-                $select->where('store_id IN (?)', $storeIds);
+                $select->where($whereCondition.' IN (?)', $storeIds);
             }
         } else {
             /**
              * For empty result
              */
-            $select->where('store_id < ?', 0);
+            $select->where($whereCondition.' < ?', 0);
         }
 
         return $select;
