diff --git a/vendor/magento/module-sales/Model/Order/Creditmemo/Validation/QuantityValidator.php b/vendor/magento/module-sales/Model/Order/Creditmemo/Validation/QuantityValidator.php
index 0a762235..fd3f6321 100644
--- a/vendor/magento/module-sales/Model/Order/Creditmemo/Validation/QuantityValidator.php
+++ b/vendor/magento/module-sales/Model/Order/Creditmemo/Validation/QuantityValidator.php
@@ -133,7 +133,7 @@ class QuantityValidator implements ValidatorInterface
         }
         $orderItem = $orderItemsById[$item->getOrderItemId()];
 
-        if (!$this->isValidDecimalRefundQty($orderItem->getIsQtyDecimal(), $item->getQty())) {
+        if (!$this->isValidDecimalRefundQty($orderItem->getIsQtyDecimal(), (float)$item->getQty())) {
             return __(
                 'We found an invalid quantity to refund item "%1".',
                 $orderItem->getSku()
                 